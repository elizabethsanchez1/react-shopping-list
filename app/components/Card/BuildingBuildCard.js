import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View, StyleSheet } from 'react-native';
import { Card as CardComponent } from 'react-native-elements';
import BuildingBuildTable from "../Table/BuildingBuildTable";
import { Input } from '../Form';
import { PrimaryButton, Link } from '../Button';
import theme from "../../styles/theme.style";

const styles = StyleSheet.create({
  titleStyle: {
    fontSize: theme.FONT_SIZE_HEADERBAR,
    fontFamily: theme.PRIMARY_FONT_FAMILY,
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
    color: theme.PRIMARY_FONT_COLOR,
    marginBottom: 0
  },

  basicCard: {
    borderRadius: 4,
    backgroundColor: theme.SECONDARY_BACKGROUND,
    borderWidth: 0,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 3 },
    shadowOpacity: 0.8,
    shadowRadius: 2,
    elevation: 1,
  },
});

const BuildingBuildCard = ( { day, updateDay, exercises, updateField, customSet, checkIfCustom, sortLink, addExercises, deleteExercises }) => {

  return (
    <CardComponent
      titleStyle={ styles.titleStyle }
      dividerStyle={ { display: 'none' } }
      containerStyle={ [ styles.basicCard, { marginTop: 30 } ] }
    >
      <Input
        defaultValue={ day }
        onEndEditing={ ( text ) => updateDay( text.nativeEvent.text ) }
        clearTextOnFocus
        autoCapitalize="words"
        // placeholderTextColor='#A1A1A1'
      />

      <BuildingBuildTable
        items={ exercises }
        updateField={ update => updateField( update ) }
        customSet={ index => customSet( index ) }
        checkIfCustom={ ( exercise, exerciseSelected ) => checkIfCustom(
            exercise,
            exerciseSelected
          )
        }
      />

      {
        ( exercises.length !== 0 )
          && <View
              style={{
                alignItems: 'flex-start',
                marginBottom: 10,
              }}
            >
              <Link
                title={ 'Sort Exercises' }
                onPress={ sortLink }
              />
            </View>
      }

      <View
        style={{
          flex: 1,
          flexDirection: 'row',
          justifyContent: 'space-between',
        }}
      >
        <PrimaryButton
          title="ADD ITEM"
          buttonStyle={ { padding: 15 } }
          containerViewStyle={ { marginLeft: 0, marginRight: 0 } }
          onPress={ () => addExercises() }
        />
        <PrimaryButton
          title="DELETE ITEM"
          buttonStyle={ { padding: 15 } }
          containerViewStyle={ { marginLeft: 0, marginRight: 0 } }
          onPress={ () => deleteExercises() }
        />
      </View>
    </CardComponent>
  );
};

BuildingBuildCard.propTypes = {
  day: PropTypes.string,
  updateDay: PropTypes.func,
  exercises: PropTypes.array,
  updateField: PropTypes.func,
  customSet: PropTypes.func,
  checkIfCustom: PropTypes.func,
  sortLink: PropTypes.func,
  addExercises: PropTypes.func,
  deleteExercises: PropTypes.func,
};

export default BuildingBuildCard;
