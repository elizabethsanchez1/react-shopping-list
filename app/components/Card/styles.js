import { create } from 'react-native-platform-stylesheet';
import theme from '../../styles/theme.style';

export default create({

  // Shared Styling
  titleStyle: {
    fontSize: theme.FONT_SIZE_HEADERBAR,
    fontFamily: theme.PRIMARY_FONT_FAMILY,
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
    color: theme.PRIMARY_FONT_COLOR,
    marginBottom: 0
  },

  basicCard: {
    borderRadius: 4,
    backgroundColor: theme.SECONDARY_BACKGROUND,
    borderWidth: 0,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 3 },
    shadowOpacity: 0.8,
    shadowRadius: 2,
    elevation: 1,
  },



  // LogsSelectedLogsCard component styling
  container: {
    marginBottom: 5,
  },

  inputView: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },

  inputSubView: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },

  inchesLabel: {
    fontFamily: theme.PRIMARY_FONT_FAMILY,
    fontSize: 30,
    minWidth: 20,
    height: 20,
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
    color: theme.PRIMARY_FONT_COLOR,
  },

  inputStyles: {
    marginRight: 20,
    borderBottomWidth: 1,
    borderBottomColor: theme.BORDER_COLOR,
    color: theme.PRIMARY_FONT_COLOR,
    fontFamily: theme.SECONDARY_FONT_FAMILY,
    fontSize: theme.FONT_SIZE_MEDIUM,
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
    minWidth: 75,
  },

  text: {
    fontFamily: theme.PRIMARY_FONT_FAMILY,
    fontSize: theme.FONT_SIZE_MEDIUM,
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
    color: theme.PRIMARY_FONT_COLOR,
    minWidth: 20
  },

  // SavedWorkoutCard Component styling

  workoutCardHeader: {
    fontFamily: theme.PRIMARY_FONT_FAMILY,
    fontSize: theme.FONT_SIZE_LARGE,
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
    color: theme.PRIMARY_FONT_COLOR,
    marginBottom: 30,
  }


});
