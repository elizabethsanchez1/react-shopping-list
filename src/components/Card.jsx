import React from 'react';
import Subtitle from './Subtitle';
import Input from './Input';
import Add from './Add';
import Search from './Search';
import Clear from './Clear';
import ListItems from './ListItems';

const styles = {
	container: {
		border: '3px solid #9099A2',
		borderRadius: 10,
		background: '#96858F',
		padding: 10,
		fontSize: 15,
		maxWidth: 500,
		margin: 'auto',
		display: 'flex',
		flexDirection: 'column',
		flexWrap: 'wrap'
	},
	centerInput: {
		margin: '0 auto',
	},
	allBtns: {
		display: 'flex',
		justifyContent: 'center',
	}
};

function Card() {
  return (
    <div style={ styles.container }>
      
      <div styles={ styles.centerInput }>
			<Subtitle />
     		<Input />
      </div> 
     
      <div styles={ styles.allBtns }>
      	<Add/>
      	<Search/>
      	<Clear/>
      </div>
      	<ListItems/>

    </div> 
  )
}

export default Card;