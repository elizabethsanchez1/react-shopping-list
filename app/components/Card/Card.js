import PropTypes from 'prop-types';
import React from 'react';
import { Card as CardComponent } from 'react-native-elements';
import styles from './styles';

const Card = (props) => {
  return (
    <CardComponent
      titleStyle={styles.titleStyle}
      dividerStyle={{ display: 'none' }}
      containerStyle={[styles.basicCard, props.containerStyling]}
      {...props}
    >
      {props.children}
    </CardComponent>
  );
};

Card.propTypes = {
  children: PropTypes.any,
  title: PropTypes.string,
  containerStyling: PropTypes.object,
};

export default Card;
