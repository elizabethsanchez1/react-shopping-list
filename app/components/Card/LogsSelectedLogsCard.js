import PropTypes from 'prop-types';
import React from 'react';
import { View, Text, TextInput } from 'react-native';
import { Card as CardComponent } from 'react-native-elements';
import styles from './styles';

const LogsSelectedLogsCard = ( { title, label, value, inputChanged } ) => {
  return (
    <CardComponent containerStyle={ [styles.basicCard, styles.container] }>
      <View style={ styles.inputView }>
        <Text style={ styles.text }>{ title }</Text>
        <View style={ styles.inputSubView }>
          <TextInput
            keyboardType='decimal-pad'
            textAlign='center'
            style={ styles.inputStyles }
            defaultValue={ value }
            onEndEditing={ ( text ) => inputChanged( text.nativeEvent.text ) }
            returnKeyType="done"
          />
          {
            ( label === '"' )
              ? <Text style={ styles.inchesLabel }>{ label }</Text>
              : <Text style={ styles.text }>{ label }</Text>
          }
        </View>
      </View>
    </CardComponent>
  );
};


LogsSelectedLogsCard.propTypes = {
  containerStyling: PropTypes.object,
  title: PropTypes.string,
  label: PropTypes.string,
  inputChanged: PropTypes.func,
};


export default LogsSelectedLogsCard;
