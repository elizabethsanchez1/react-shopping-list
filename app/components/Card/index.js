import Card from './Card';
import LogsSelectedLogsCard from './LogsSelectedLogsCard';
import SavedWorkoutCard from './Shared/SavedWorkoutCard';
import BuildingBuildCard from './BuildingBuildCard';

export { Card, LogsSelectedLogsCard, SavedWorkoutCard, BuildingBuildCard };
