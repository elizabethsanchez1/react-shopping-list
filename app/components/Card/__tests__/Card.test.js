import 'react-native';
import React from 'react';
import { Text } from 'react-native';
import Card from './../Card';
import renderer from 'react-test-renderer';


describe('Card Component', () => {


  it('Card base visually looks the same', () => {
    const tree = renderer.create(
      <Card
        title="title renders"
      >
        <Text>Testing</Text>
      </Card>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  })


});




