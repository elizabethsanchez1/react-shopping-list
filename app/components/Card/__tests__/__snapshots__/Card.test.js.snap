// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component Card base visually looks the same 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "margin": 15,
        "marginBottom": 0,
        "padding": 15,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      },
      undefined,
      Array [
        Object {
          "backgroundColor": "#47474D",
          "borderRadius": 4,
          "borderWidth": 0,
          "elevation": 1,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        undefined,
      ]
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#43484d",
                "fontSize": 17.5,
                "fontWeight": "bold",
                "marginBottom": 15,
                "textAlign": "center",
              },
              undefined,
              Object {
                "color": "#FFF",
                "fontFamily": "RobotoCondensed-Regular",
                "fontSize": 20,
                "fontWeight": "600",
                "marginBottom": 0,
              },
              undefined,
            ],
          ]
        }
      >
        title renders
      </Text>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#e1e8ee",
              "height": 1,
            },
            Array [
              Object {
                "marginBottom": 15,
              },
              Object {
                "display": "none",
              },
            ],
          ]
        }
      />
    </View>
    <Text>
      Testing
    </Text>
  </View>
</View>
`;

exports[`Card Component matches the snapShot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "margin": 15,
        "marginBottom": 0,
        "padding": 15,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      },
      undefined,
      Array [
        Object {
          "backgroundColor": "#47474D",
          "borderRadius": 4,
          "borderWidth": 0,
          "elevation": 1,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
        undefined,
      ]
    }
  />
</View>
`;
